src/xdist/workermanage.py

```diff
@@ -41,13 +41,15 @@ def __init__(self, config, specs=None, defaultchdir="pyexecnetcache") -> None:
         self.testrunuid = self.config.getoption("testrunuid")
         if self.testrunuid is None:
             self.testrunuid = uuid.uuid4().hex
-        self.group = execnet.Group()
+        self.group = execnet.Group(execmodel="main_thread_only")
```

Comment:
Yeah, seems much safer to use an explicit execmodel argument everywhere.

Reply:
Note that the main_thread_only execmodel raises a deadlock error to the channel in order to compensate for lack of back pressure when concurrent remote_exec requests are submitted, since local remote_exec calls do not trigger blocking on the local side when the main thread is busy on the remote side.